avldb/src/main/scala/scorex/db/LDBFactory.scala

```diff
@@ -0,0 +1,128 @@
+
+import org.iq80.leveldb.{DB, Range, DBFactory, DBIterator, Options, ReadOptions, Snapshot, WriteBatch, WriteOptions}
+import scorex.util.ScorexLogging
+
+import scala.collection.mutable
+import scala.util.Try
+
+case class StoreRegistry(val factory : DBFactory) extends DBFactory {
+
+  val lock = new ReentrantReadWriteLock()
+  val map = new mutable.HashMap[File, RegisteredDB]
+
+  case class RegisteredDB(val impl:DB, val path: File) extends DB {
+    var count: Int = 0
+
+    def get(key: Array[Byte]): Array[Byte] = impl.get(key)
+
+    def get(key: Array[Byte], options: ReadOptions): Array[Byte] = impl.get(key, options)
+
+    def iterator: DBIterator = impl.iterator
+
+    def iterator(options: ReadOptions): DBIterator = impl.iterator(options)
+
+    def put(key: Array[Byte], value: Array[Byte]) = impl.put(key, value)
+
+    def delete(key: Array[Byte]) = impl.delete(key)
+
+    def write(batch: WriteBatch) = impl.write(batch)
+
+    def write(batch: WriteBatch, options: WriteOptions) = impl.write(batch, options)
+
+    def createWriteBatch: WriteBatch = impl.createWriteBatch()
+
+    def put(key: Array[Byte], value: Array[Byte], options: WriteOptions) = impl.put(key, value, options)
+
+    def delete(key: Array[Byte], options: WriteOptions) = impl.delete(key, options)
+
+    def getSnapshot: Snapshot = impl.getSnapshot()
+
+    def getApproximateSizes(ranges: Range*): Array[Long] = impl.getApproximateSizes(ranges: _*)
+
+    def getProperty(name: String): String = impl.getProperty(name)
+
+    def suspendCompactions = impl.suspendCompactions()
+
+    def resumeCompactions = impl.resumeCompactions()
+
+    def compactRange(begin: Array[Byte], end: Array[Byte]) = impl.compactRange(begin, end)
+
+    override def close() = {
+      remove(path)
+      impl.close()
+    }
+  }
+
+  private def add(file: File, create: => DB): DB = {
+    lock.writeLock().lock()
+    try {
+      map.getOrElseUpdate(file, new RegisteredDB(create, file))
+    } finally {
+      lock.writeLock().unlock()
+    }
+  }
+
+  private def remove(path:File): Unit = {
+    lock.writeLock().lock()
+    try {
+      map.remove(path)
+    } finally {
+      lock.writeLock().unlock()
+    }
+  }
+
+  def open(path: File, options: Options): DB = {
+    lock.writeLock().lock()
+    try {
+      add(path, factory.open(path, options))
+    } finally {
+      lock.writeLock().unlock()
+    }
+  }
+
+  def destroy(path: File, options: Options) = {
+    factory.destroy(path, options)
+  }
+
+  def repair(path: File, options: Options) = {
+    factory.repair(path, options)
+  }
+}
+
+object LDBFactory extends ScorexLogging {
+
+  private val nativeFactory = "org.fusesource.leveldbjni.JniDBFactory"
+  private val javaFactory   = "org.iq80.leveldb.impl.Iq80DBFactory"
+
+  lazy val factory: DBFactory = {
+    val loaders = List(ClassLoader.getSystemClassLoader, this.getClass.getClassLoader)
+    val factories = List(nativeFactory, javaFactory)
```

Comment:
@kushti @oskin1 I suggest to remove `javaFactory` from this list.

Reply:
As far as I remember, the presence of experimental Java implementation of leveldb was motivated by the fact that not all target platforms have leveldb installed.
If we are not going to include leveldb binaries in our distributive, then we should allow to use Java leveldb port in this case.

Reply:
We are already including leveldb binaries into ergo jar. 
But java version should be left here as fallback implementation. It already helped guys who ran the node on rare platform so there weren’t native leveldb binaries available for it.

Reply:
@oskin1 ok, but we know Java version is unstable, so it has its own risks. Then we need to add warning messages to log with big exclamation marks and disclaimer, that it is not supported and experimental feature.